i3wsgroups/i3_workspace_groups.py

```diff
@@ -364,6 +364,22 @@ def get_tree(self, cached: bool = True) -> i3ipc.Con:
         self.tree = self.i3_connection.get_tree()
         return self.tree
 
+    def get_workspaces(self) -> List[i3ipc.Con]:
+        name_to_workspace = {}
+        for workspace in self.get_tree().workspaces():
+            name_to_workspace[workspace.name] = workspace
+        workspaces = []
+        for ws_display_metadata in self.get_workspaces_display_metadata():
+            ws_name = ws_display_metadata.workspace_name
+            if ws_name == _SCRATCHPAD_WORKSPACE_NAME:
+                continue
+            if ws_name not in name_to_workspace:
+                logger.warning('Unknown workspace detected: %s', ws_name)
+                continue
+            workspace = name_to_workspace[ws_name]
+            workspaces.append(workspace)
+        return workspaces
+
```

Comment:
Logic is duplicated from `_get_monitor_to_workspaces` - may want to factor out what is shared.